### PR TITLE
[bitnami/keycloak] Add support for proxy-headers

### PR DESCRIPTION
**Pull request by Kajot-dev**
### Description of the change

This PR implements new proxy setting introduced in Keycloak. It's a helm chart equivalent to https://github.com/bitnami/containers/pull/67957 and is meant to be used together

### Benefits

Not using deprecated options

### Applicable issues

- fixes #65190

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/keycloak/values.yaml [hunk 1]
```diff
@@ -211,7 +211,11 @@ adminRealm: "master"
 ## @param production Run Keycloak in production mode. TLS configuration is required except when using proxy=edge.
 ##
 production: false
+## @param proxyHeaders Set Keycloak proxy headers
+##
+proxyHeaders: ""
 ## @param proxy reverse Proxy mode edge, reencrypt, passthrough or none
+## DEPRECATED: use proxyHeaders instead
```
Comment (juan131) [inline bitnami/keycloak/values.yaml:218]: We should set the value to an empty string by default, right?

File: bitnami/keycloak/templates/statefulset.yaml [hunk 2]
```diff
@@ -217,11 +217,11 @@ spec:
             {{- end }}
             {{- if .Values.adminIngress.enabled }}
             - name: KC_HOSTNAME_ADMIN_URL
-              value: "http{{ if or .Values.adminIngress.tls (eq .Values.proxy "edge") }}s{{ end }}://{{ include "common.tplvalues.render" (dict "value" .Values.adminIngress.hostname "context" $) }}"
+              value: "http{{ if or .Values.adminIngress.tls (eq .Values.proxy "edge") (not (empty .Values.proxyHeaders)) }}s{{ end }}://{{ include "common.tplvalues.render" (dict "value" .Values.adminIngress.hostname "context" $) }}"
```
Comment (juan131) [inline bitnami/keycloak/templates/statefulset.yaml:220]: good catch

**Review by juan131 — COMMENTED** [review]
Thanks for the PR! Please check my comments

**Comment by juan131** [comment]
@Kajot-dev we released a new container image version (see `24.0.5-debian-12-r3`) including the associated changes in the image. Please update your branch with latest changes in `main`

File: bitnami/keycloak/values.yaml (hunk 1 above)
Comment (Kajot-dev) [reply bitnami/keycloak/values.yaml:218]: You mean for the deprecated `proxy` value? We can, but it'll be a breaking change

Comment (Kajot-dev) [reply bitnami/keycloak/values.yaml:218]: Done

**Comment by Kajot-dev** [comment]
@juan131 Changed the default value of `proxy` to `""` like you asked.

Rebased against latest main

**Review by juan131 — APPROVED** [review]
LGTM

**Comment by juan131** [comment]
@Kajot-dev could you update the image tag in the **values.yaml** to use `24.0.5-debian-12-r3`? Otherwise the CI will fail.

**Review by juan131 — DISMISSED** [review]
LGTM

**Review by juan131 — DISMISSED** [review]

**Review by juan131 — APPROVED** [review]

**Comment by kedare** [comment]
I think this should have been retro-compatible ?

Before I could deploy the chart with just `proxy=edge` and now it's failing as `proxyHeaders` is empty.

@juan131 @Kajot-dev

**Comment by aqeelat** [comment]
I'm in the same boat.
I have:
```yaml
production: true
proxy: edge
```
and now I don't know what to do.

**Comment by Kajot-dev** [comment]
> I think this should have been retro-compatible ?
> 
> Before I could deploy the chart with just `proxy=edge` and now it's failing as `proxyHeaders` is empty.
> 
> @juan131 @Kajot-dev 

I can fix it by the end of the week

**Comment by Kajot-dev** [comment]
> I'm in the same boat.
> I have:
> ```yaml
> production: true
> proxy: edge
> ```
> and now I don't know what to do.
> 
> The diff I'm getting in keycloak-env-vars is:
> ```diff
> - KEYCLOAK_PROXY: edge
> + KEYCLOAK_PROXY_HEADERS: forwarded|xforwarded
> ```

The diff seems good. Are you using the new image?

**Comment by Kajot-dev** [comment]
@man0s @aqeelat @kedare Please see mentioned PR (#28530) if it fixes the issue and please provide some more info. What is happening? Chart does not template? Container does not start? (if so, then provide logs)

**Pull request merged**